### PR TITLE
Remove relay SSR

### PR DESCRIPTION
**Pull request by dleard**
This PR has removed SSR from relay, which fixes the react navigation tabs.
Relay SSR appears to be somewhat broken at the moment with active Issues and PRs:

https://github.com/facebook/relay/issues/1881
https://github.com/facebook/relay/pull/2883

### CODE REVIEW

File: app/lib/relay-environment/server.ts [hunk 1]
```diff
@@ -2,42 +2,35 @@ import {
...
             // url: req => process.env.RELAY_ENDPOINT
             url: _ => 'http://localhost:3004/graphql'
-          }),
-          relaySSR.getMiddleware()
+          })
         ])
       })
     };
   },
-  createEnvironment: (relayData, key) => {
```
Comment (matthieu-foucault) [inline app/lib/relay-environment/server.ts]: we're not passing any `relayData` anymore when creating the server environment

File: app/pages/_app.tsx [hunk 2]
```diff
@@ -47,19 +49,20 @@ export default class App extends NextApp<AppProps> {
...
+      pageProps: {variables = {}}
     } = this.props;
     const environment = createEnvironment(
-      relayData,
       JSON.stringify({
         queryID: Component.query
           ? getRequest(Component.query).params.name
           : undefined,
-        variables
+        variables: {...variables, ...router.query}
```
Comment (matthieu-foucault) [inline app/pages/_app.tsx]: Should this parameter be removed as well?

File: app/pages/_app.tsx [hunk 3]
```diff
@@ -47,19 +49,20 @@ export default class App extends NextApp<AppProps> {
...
       })
     );
     return (
       <ErrorBoundary>
+        {/*
+          // @ts-ignore */}
```
Comment (matthieu-foucault) [inline app/pages/_app.tsx]: Do we need this?

File: app/pages/_app.tsx (hunk 2 above)
Comment (dleard) [reply app/pages/_app.tsx]: need this one

File: app/lib/relay-environment/server.ts (hunk 1 above)
Comment (dleard) [reply app/lib/relay-environment/server.ts]: right! I've removed it

File: app/pages/_app.tsx (hunk 3 above)
Comment (dleard) [reply app/pages/_app.tsx]: nope!

File: app/pages/_app.tsx (hunk 2 above)
Comment (dleard) [reply app/pages/_app.tsx]: It's used in the client.ts fetchQuery function to determine whether the same query with the same variables is in the cache and to just use the cache in that case. This is also why mutations clear the cache.

**Pull request merged**